Stratified K-Fold Cross Validation

SeanLF opened the pull request:
Closes #31

SeanLF committed:
Cross validation

SeanLF commented on active_learning/learn.py:
```diff
@@ -0,0 +1,137 @@
+        # test model
+        predicted = trained_model.predict(X_test)
+        results.append({'y_true': y_test, 'y_pred': predicted})
+        print('Cross val iteration done\n')
```
To remove

SeanLF committed:
remove print statement

markkasun commented on active_learning/learn.py:
```diff
@@ -0,0 +1,136 @@
+    model = SGDClassifier(loss='hinge', penalty='l2', alpha=1e-2, n_iter=1000, random_state=None, learning_rate='optimal', class_weight='balanced')
+    # train on labeled instances
+    model.fit(*(training_dataset.format_sklearn()))
+    # active learning
```
Try to reuse the code from learning.py. Create a separate method with condition statements that determine whether to produce intermediate results

markkasun commented on active_learning/learn.py:
```diff
@@ -0,0 +1,136 @@
+    results = []
+    # make K folds
+    skf = StratifiedKFold(n_splits=num_folds)
+    skf.get_n_splits(X, y)
```
Remove this line. Does nothing.

markkasun commented on active_learning/learn.py:
```diff
@@ -0,0 +1,136 @@
+
+def learn():
+    # Build dataset
+    dataset_name = 'Four University'
```
Should be a parameter to be able to run on other datasets. Same with other variables below.

SeanLF committed:
Resolve 2/3 PR requested changes

SeanLF committed:
train_active_learning param for intermediate results

SeanLF committed:
resolve rebase issues

SeanLF committed:
Refactor active learning model training

- rename custom labelers
- refactor learn function to re-use the method to train the active
learning in learn.py
- add missing functionality into learn.train_active_learning from
learning. learn

markkasun commented on active_learning/learn.py:
```diff
@@ -79,7 +85,7 @@ def learn(**kwargs):
     num_queries = kwargs.pop('num_queries', 20)
     num_folds = kwargs.pop('num_folds', 10)
-    labeled_instances = kwargs.pop('dataset_id', 1)
+    labeled_instances = kwargs.pop('dataset_id', 5)
```
5?

SeanLF replied:
oh yeah, my bad

markkasun commented on active_learning/learning.py:
```diff
@@ -101,43 +101,28 @@ def learn(self, **kwargs):
 
-        train_dataset, train_dataset_labeled, test_dataset, lookup_train, lookup_test, self.lookup_table = \
-            self.setup_datasets(X, self.y, self.lookup_table, train_size)
+        train_dataset, train_dataset_labeled, test_dataset, lookup_train, lookup_test, self.lookup_table = self.setup_datasets(X, self.y, self.lookup_table, train_size)
```
Don't like the split lines? Much easier to read imo.

SeanLF replied:
nope, python multiline statements are gross 😛

markkasun replied:
But think of the PEP8!

SeanLF replied:
I increased the max line length... haha

markkasun commented on active_learning/custom_labelers.py:
```diff
@@ -62,3 +64,18 @@ def connect_handler(self, data):
+    Uses the ground truth to label instances
+    """
+
+    def __init__(self, **kwargs):
```
Should make oracle_dataset a mandatory field?
     if self.oracle_dataset is None:
            raise TypeError(
                "__init__() missing required keyword-only argument: 'oracle_dataset'"
            )

SeanLF replied:
might have to move my params around

SeanLF committed:
resolve PR code review issues
